api/python/t4/session.py

```diff
@@ -190,6 +190,10 @@ def login_with_token(refresh_token):
 
     clear_session()
 
+    # use registry-provided credentials if available
+    if try_registry_credentials():
```

Comment:
This seems like a bad idea - I think we should make it explicit. Also, this could fail due to things like network errors.

Reply:
But it's still here?